src/compiler/crystal/tools/doc/markd_doc_renderer.cr

```diff
@@ -74,9 +74,9 @@ class Crystal::Doc::MarkdDocRenderer < Markd::HTMLRenderer
 
     # Check Type#method(...) or Type or #method(...)
     text.gsub %r(
-      ((?:\B::)?\b[A-Z]\w+(?:\:\:[A-Z]\w+)*|\B|(?<=\bself))([#.])([\w<=>+\-*\/\[\]&|?!^~]+[?!]?)(?:\((.*?)\))?
+      ((?:\B::)?\b[A-Z]\w*(?:\:\:[A-Z]\w*)*|\B|(?<=\bself))([#.])([\w<=>+\-*\/\[\]&|?!^~]+[?!]?)(?:\((.*?)\))?
         |
-      ((?:\B::)?\b[A-Z]\w+(?:\:\:[A-Z]\w+)*)
+      ((?:\B::)?\b[A-Z]\w*(?:\:\:[A-Z]\w*)*)
```

Comment:
I don't think I fully comprehend the intention of this regex, neither before nor after this change.
What's the purpose of matching for double colon followed by a letter and (now optional) whitespace? 🤔

Reply:
I think maybe the main confusion is that `\w` is "word character", not "whitespace" as I'm guessing that you assumed 😅

Reply:
Yep. Doh 🤦